fetch/fetch.go

```diff
@@ -268,9 +269,13 @@ func NewFetch(
 	proposals *store.Store,
 	host *p2p.Host,
 	opts ...Option,
-) *Fetch {
+) (*Fetch, error) {
 	bs := datastore.NewBlobStore(cdb, proposals)
 
+	hashPeerCache, err := NewHashPeersCache(cacheSize)
+	if err != nil {
+		return nil, fmt.Errorf("create hash peer cache: %w", err)
```

Comment:
IMHO, propagating this up isn't the right choice. It can only fail if the private `cacheSize` const is invalid so the problem is:
- a clear programmer error
- entirely isolated to this pkg/file

I think it's totally fine to panic on it rather than complicating other code with an impossible (in normal conditions, when `cacheSize >=0` invariant isn't violated) error, wdyt?